### PR TITLE
Straight pp

### PR DESCRIPTION
**Pull request by kevinatorchen**
This pull request would close issue #66 in RRT.

[edit by @jgkamat] closes robojackets/rrt#66

### CODE REVIEW

File: soccer/planning/RRTPlanner.cpp [hunk 1]
```diff
@@ -88,6 +91,9 @@ std::unique_ptr<Path> RRTPlanner::run(PlanRequest& planRequest) {
     vector<DynamicObstacle> actualDynamic;
     splitDynamic(obstacles, actualDynamic, dynamicObstacles);
 
+    //Straight line path planning
```
Comment (jgkamat) [inline soccer/planning/RRTPlanner.cpp]: This comment can probably be removed.

File: soccer/planning/RRTPlanner.cpp [hunk 2]
```diff
@@ -192,7 +215,11 @@ vector<Point> RRTPlanner::runRRT(MotionInstant start, MotionInstant goal,
     biRRT.setStepSize(*RRTConfig::StepSize);
     biRRT.setMinIterations(_minIterations);
     biRRT.setMaxIterations(_maxIterations);
```
Comment (jgkamat) [inline soccer/planning/RRTPlanner.cpp]: Right now, this runs the RRT twice for every path plan, can we bump down minIterations to 0, and decrease the max iterations, and maybe even bump up the stepsize to get the first iteration as fast as possible? I think we might even be able to get it to only take one step (which would be much faster than running it twice). 

Make sure to set these dynamically, for example, don't hard code the stepsize higher, you could just set it to the distance between the goal and the target (and maybe you could set minIterations to 2).

File: soccer/planning/RRTPlanner.cpp [hunk 3]
```diff
@@ -177,13 +184,29 @@ std::unique_ptr<InterpolatedPath> RRTPlanner::generateRRTPath(
...
+    vector<Point> straight = runRRTHelper(start, goal, motionConstraints, obstacles, state, shellID, true);
+    if (straight.size() != 0) {
+        cout << "Straight path!" << endl;
+        return straight;
+    }
+    cout << "Regular RRT Planning" << endl;
+    return runRRTHelper(start, goal, motionConstraints, obstacles, state, shellID, false);
+}
+
+vector<Point> RRTPlanner::runRRTHelper(MotionInstant start, MotionInstant goal,
```
Comment (jgkamat) [inline soccer/planning/RRTPlanner.cpp]: Can you add a documentation header to this method?

Comment (kevinatorchen) [inline soccer/planning/RRTPlanner.cpp]: There is already a Javadoc for this method in the header file. It reads:

/**
     * Helper method for runRRT(), which creates a vector of points representing 
     * the RRT path.
     */

Comment (jgkamat) [inline soccer/planning/RRTPlanner.cpp]: oh, that slipped my mind, sorry, your good! :smile:

File: soccer/planning/RRTPlanner.cpp [hunk 4]
```diff
@@ -6,12 +6,14 @@
...
 #include "RoboCupStateSpace.hpp"
 #include "motion/TrapezoidalMotion.hpp"
+#include "DirectTargetPathPlanner.hpp"
 
 #include <stdio.h>
 #include <stdlib.h>
 #include <Eigen/Dense>
 #include <algorithm>
 #include <iostream>
+#include <sys/time.h>
```
Comment (jgkamat) [inline soccer/planning/RRTPlanner.cpp]: Can this be removed :P

**Comment by jgkamat** [comment]
I think the latest commit reverted some updated submodules, I'll bump them back for now, so builds can pass again. Make sure to either run `git submodule update` or avoid commiting submodule changes when pulling new changes!

File: soccer/planning/RRTPlanner.cpp [hunk 5]
```diff
@@ -6,6 +6,7 @@
 #include "RRTUtil.hpp"
 #include "RoboCupStateSpace.hpp"
 #include "motion/TrapezoidalMotion.hpp"
+#include "DirectTargetPathPlanner.hpp"
```
Comment (jgkamat) [inline soccer/planning/RRTPlanner.cpp]: Is this include being used anywhere?

**Comment by kevinatorchen** [comment]
Hey Jay, I have removed the unnecessary import statement, and I have modified the call to biRRT's getPath() method because the way the method returns the value is now different. (Instead of directly returning the value, it now uses an out parameter.)

**Comment by jgkamat** [comment]
Thanks! This looks ready to merge, thanks!

I'll pull this in a bit.

**Pull request merged**